src/type_aliases.rs

```diff
@@ -2,19 +2,29 @@
 
 //! Publicly exported type aliases.
 
+use crate::{
+    internal::incompatibility::IncompId, solver::DependencyProvider, version_set::VersionSet,
+};
+
 /// Map implementation used by the library.
 pub type Map<K, V> = rustc_hash::FxHashMap<K, V>;
 
 /// Set implementation used by the library.
 pub type Set<V> = rustc_hash::FxHashSet<V>;
 
+/// The version type associated with the version set associated with the dependency provider.
+pub type V<DP> = <<DP as DependencyProvider>::VS as VersionSet>::V;
```

Comment:
That alias makes my brain hurt ahah. Do we need the `<DP>` when the V is defined within a dependency provider?

Reply:
Oof. I also found this a bit awkward but don't know if I have an alternative. I recall that putting `V` in the `DependencyProvider` is complicated too?

Reply:
I had a commit where `V` was a associated type on `DP`. It made the use within the library a lot easier. But it required users to confirm that `type V = VS::V`, which ended up being ugly and tricky. I unfortunately drop that commit when rebasing. I will attempt to reconstruct it so we can decide which poison is worse.

Reply:
`git reflog` did its magic and I was able to find the old commit! And looking it over maybe it's not so bad. What you all think?